guard/resources/validate/functions/rules/join_with_message.guard

```diff
@@ -5,5 +5,8 @@ rule TEST_COLLECTION when %template !empty {
 
     let res = join(%collection, ",")
     %res == "a,b"
+
+    "a,b" == join(%collection, ",")
```

Comment:
This is a good example. We should try to simplify other examples and make it as easy to read as this is.